code/__defines/tgs.dm

```diff
@@ -0,0 +1,35 @@
+// Required interfaces (fill in with your codebase equivalent):
+#define TGS_EXTERNAL_CONFIGURATION
+
+/// Create a global variable named `Name` and set it to `Value`.
+#define TGS_DEFINE_AND_SET_GLOBAL(Name, Value) var/global/##Name = ##Value
+
+/// Read the value in the global variable `Name`.
+#define TGS_READ_GLOBAL(Name) global.##Name
+
+/// Set the value in the global variable `Name` to `Value`.
+#define TGS_WRITE_GLOBAL(Name, Value) global.##Name = ##Value
+
+/// Disallow ANYONE from reflecting a given `path`, security measure to prevent in-game use of DD -> TGS capabilities.
+#define TGS_PROTECT_DATUM(Path) //GENERAL_PROTECT_DATUM(##Path)
```

Comment:
The things you can do with direct var read/call access to the TGS datum are fairly benign across the board (you can't really do much remotely even if you have the secret api key of a given session), but it'd be worth looking into a protect datum define if it isn't already present.